File: hello-php/src/Client/AudioClientApplication.php
<?php

namespace AudioStreamClient;

use AudioStreamClient\Core\WebSocketClient;
use AudioStreamClient\Core\FileManager;
use AudioStreamClient\Core\UploadManager;
use AudioStreamClient\Core\DownloadManager;
use AudioStreamClient\Util\PerformanceMonitor;
use AudioStreamClient\Util\VerificationModule;

/**
 * Audio client application entry point.
 * Coordinates upload, download, and verification operations.
 */
class AudioClientApplication
{
    private array $config;
    private PerformanceMonitor $performance;

    /**
     * Create a new AudioClientApplication.
     *
     * @param array $config Configuration array with inputPath, outputPath, serverUri, verbose
     */
    public function __construct(array $config)
    {
        $this->config = $config;
        $this->performance = new PerformanceMonitor();
    }

    /**
     * Run the client application.
     *
     * @return int Exit code (0 for success, 1 for failure)
     */
    public function run(): int
    {
        try {
            Logger::setVerbose($this->config['verbose']);
            
            Logger::info('Audio Stream Client');
            Logger::info('Input: ' . $this->config['inputPath']);
            Logger::info('Output: ' . $this->config['outputPath']);
            Logger::info('Server: ' . $this->config['serverUri']);

            // Initialize performance monitor
            $fileSize = FileManager::getFileSize($this->config['inputPath']);
            $this->performance->setFileSize($fileSize);

            // Connect to WebSocket server
            $ws = new WebSocketClient($this->config['serverUri']);
            $ws->connect();
            
            // Upload file
            $this->performance->startUpload();
            $streamId = UploadManager::upload($ws, $this->config['inputPath']);
            $this->performance->endUpload();
            
            // Sleep 2 seconds after upload
            Logger::info('Upload successful, sleeping for 2 seconds...');
            usleep(2000000); // 2 seconds in microseconds
            
            // Download file
            $this->performance->startDownload();
            DownloadManager::download($ws, $streamId, $this->config['outputPath'], $fileSize);
            $this->performance->endDownload();
            
            // Sleep 2 seconds after download
            Logger::info('Download successful, sleeping for 2 seconds...');
            usleep(2000000); // 2 seconds in microseconds
            
            // Verify integrity
            $verification = VerificationModule::verify($this->config['inputPath'], $this->config['outputPath']);
            
            // Report performance
            $report = $this->performance->getReport();
            $this->performance->printReport($report);
            
            // Close connection
            $ws->close();
            
            // Exit with appropriate code
            if ($verification['passed']) {
                Logger::info('SUCCESS: File transfer completed successfully');
                return 0;
            } else {
                Logger::error('FAILURE: File verification failed');
                return 1;
            }

        } catch (\Exception $e) {
            Logger::error('Fatal error: ' . $e->getMessage());
            return 1;
        }
    }
}

File: hello-dart/lib/client/audio_client_application.dart
// Main entry point for Dart audio stream client.
// Handles command-line parsing and coordinates upload/download operations.
// Matches Java AudioClientApplication functionality.

import 'dart:io';
import '../src/cli_parser.dart';
import '../src/logger.dart';
import '../src/types.dart';
import 'core/websocket_client.dart';
import 'core/upload_manager.dart';
import 'core/download_manager.dart';
import 'core/file_manager.dart';
import 'util/verification_module.dart';
import 'util/performance_monitor.dart';

/// Audio client application entry point.
class AudioClientApplication {
  /// Run the audio client application.
  static Future<void> run(List<String> arguments) async {
    try {
      final config = CliParser.parse(arguments);
      if (config == null) {
        exit(1);
      }

      await runWorkflow(config);
      exit(0);
    } catch (e) {
      Logger.error('Fatal error: $e');
      exit(1);
    }
  }

  /// Execute the upload/download workflow.
  static Future<void> runWorkflow(Config config) async {
    Logger.info('Audio Stream Client');
    Logger.info('Input: ${config.inputPath}');
    Logger.info('Output: ${config.outputPath}');
    Logger.info('Server: ${config.serverUri}');

    // Initialize performance monitor
    final performance = Performance();
    final fileSize = await FileManager.getFileSize(config.inputPath);
    performance.setFileSize(fileSize);

    // Connect to WebSocket server
    final ws = WebSocketClient(config.serverUri);
    await ws.connect();

    try {
      // Upload file
      performance.startUpload();
      final streamId = await Upload.upload(ws, config.inputPath);
      performance.endUpload();

      // Sleep 2 seconds after upload
      Logger.info('Upload successful, sleeping for 2 seconds...');
      await Future.delayed(Duration(seconds: 2));

      // Download file
      performance.startDownload();
      await Download.download(ws, streamId, config.outputPath, fileSize);
      performance.endDownload();

      // Sleep 2 seconds after download
      Logger.info('Download successful, sleeping for 2 seconds...');
      await Future.delayed(Duration(seconds: 2));

      // Verify integrity
      final verification =
          await Verification.verify(config.inputPath, config.outputPath);

      // Report performance
      final report = performance.getReport();
      performance.printReport(report);

      // Exit with appropriate code
      if (verification.passed) {
        Logger.info('SUCCESS: Stream completed successfully');
      } else {
        Logger.error('FAILURE: File verification failed');
        throw Exception('Verification failed');
      }
    } finally {
      await ws.close();
    }
  }
}

void main(List<String> arguments) async {
  await AudioClientApplication.run(arguments);
}
